daemon/cmd/daemon.go

```diff
@@ -921,9 +921,7 @@ func NewDaemon(ctx context.Context, cleaner *daemonCleanup,
 	// BPF masquerade depends on BPF NodePort and require host-reachable svc to
 	// be fully enabled in the tunneling mode, so the following checks should
 	// happen after invoking initKubeProxyReplacementOptions().
-	if option.Config.EnableIPv4Masquerade && option.Config.EnableBPFMasquerade &&
-		(!option.Config.EnableNodePort || option.Config.EgressMasqueradeInterfaces != "" || !option.Config.EnableRemoteNodeIdentity ||
-			(option.Config.TunnelingEnabled() && !hasFullHostReachableServices())) {
+	if option.Config.EnableIPv4Masquerade && option.Config.EnableBPFMasquerade {
```

Comment:
This refactoring will end up always disabling BPF masquerading and throwing a warning.

Reply:
ah, maybe add `if (msg)` then set `option.Config.EnableBPFMasquerade = false`

Reply:
I also don't need to add the switch `case option.Config.EnableHostLegacyRouting` after the refactor, since empty msg will not happen even I enable `option.Config.EnableHostLegacyRouting`